kernel/arch/dreamcast/include/arch/timer.h

```diff
@@ -225,6 +225,144 @@ int timer_init();
+    C is count, and B is time of one bus cycle).
+*/
+#define PMCR_COUNT_RATIO_CYCLES 1
+
+/*                MODE DEFINITION                  VALUE   MEASURMENT TYPE & NOTES */
+#define PMCR_INIT_NO_MODE                           0x00 /* None; Just here to be complete */
+#define PMCR_OPERAND_READ_ACCESS_MODE               0x01 /* Quantity; With cache */
+#define PMCR_OPERAND_WRITE_ACCESS_MODE              0x02 /* Quantity; With cache */
+#define PMCR_UTLB_MISS_MODE                         0x03 /* Quantity */
+#define PMCR_OPERAND_CACHE_READ_MISS_MODE           0x04 /* Quantity */
+#define PMCR_OPERAND_CACHE_WRITE_MISS_MODE          0x05 /* Quantity */
+#define PMCR_INSTRUCTION_FETCH_MODE                 0x06 /* Quantity; With cache */
+#define PMCR_INSTRUCTION_TLB_MISS_MODE              0x07 /* Quantity */
+#define PMCR_INSTRUCTION_CACHE_MISS_MODE            0x08 /* Quantity */
+#define PMCR_ALL_OPERAND_ACCESS_MODE                0x09 /* Quantity */
+#define PMCR_ALL_INSTRUCTION_FETCH_MODE             0x0a /* Quantity */
+#define PMCR_ON_CHIP_RAM_OPERAND_ACCESS_MODE        0x0b /* Quantity */
+/* No 0x0c */
+#define PMCR_ON_CHIP_IO_ACCESS_MODE                 0x0d /* Quantity */
+#define PMCR_OPERAND_ACCESS_MODE                    0x0e /* Quantity; With cache, counts both reads and writes */
+#define PMCR_OPERAND_CACHE_MISS_MODE                0x0f /* Quantity */
+#define PMCR_BRANCH_ISSUED_MODE                     0x10 /* Quantity; Not the same as branch taken! */
+#define PMCR_BRANCH_TAKEN_MODE                      0x11 /* Quantity */
+#define PMCR_SUBROUTINE_ISSUED_MODE                 0x12 /* Quantity; Issued a BSR, BSRF, JSR, JSR/N */
+#define PMCR_INSTRUCTION_ISSUED_MODE                0x13 /* Quantity */
+#define PMCR_PARALLEL_INSTRUCTION_ISSUED_MODE       0x14 /* Quantity */
+#define PMCR_FPU_INSTRUCTION_ISSUED_MODE            0x15 /* Quantity */
+#define PMCR_INTERRUPT_COUNTER_MODE                 0x16 /* Quantity */
+#define PMCR_NMI_COUNTER_MODE                       0x17 /* Quantity */
+#define PMCR_TRAPA_INSTRUCTION_COUNTER_MODE         0x18 /* Quantity */
+#define PMCR_UBC_A_MATCH_MODE                       0x19 /* Quantity */
+#define PMCR_UBC_B_MATCH_MODE                       0x1a /* Quantity */
+/* No 0x1b-0x20 */
+#define PMCR_INSTRUCTION_CACHE_FILL_MODE            0x21 /* Cycles */
+#define PMCR_OPERAND_CACHE_FILL_MODE                0x22 /* Cycles */
+#define PMCR_ELAPSED_TIME_MODE                      0x23 /* Cycles; For 200MHz CPU: 5ns per count in 1 cycle = 1 count mode, or around 417.715ps per count (increments by 12) in CPU/bus ratio mode */
+#define PMCR_PIPELINE_FREEZE_BY_ICACHE_MISS_MODE    0x24 /* Cycles */
+#define PMCR_PIPELINE_FREEZE_BY_DCACHE_MISS_MODE    0x25 /* Cycles */
+/* No 0x26 */
+#define PMCR_PIPELINE_FREEZE_BY_BRANCH_MODE         0x27 /* Cycles */
+#define PMCR_PIPELINE_FREEZE_BY_CPU_REGISTER_MODE   0x28 /* Cycles */
+#define PMCR_PIPELINE_FREEZE_BY_FPU_MODE            0x29 /* Cycles */
```

Comment:
It'd be nice to have Doxygen comments on all of these. I'll merge without for now, but if that could be done in another PR, I'd appreciate it.